woocommerce/woocommerce
# [CYS on Core] Display the theme switch modal only if the current active theme is not TT4

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Ensure the theme switch warning modal in the transitional page of the CYS is displayed exclusively for WordPress installs with a theme different than TT4.

Add a new recording event called `'customize_your_store_agree_to_theme_switch_click'` whenever the user clicks on the "Design a new theme" button.

⚠️ Further changes are planned to be implemented as part of a separate PR to identify if the TT4 theme has been modified by the user.





Closes #45281



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

## Initial Setup

1. Enable the WooCommerce Beta Tester plugin if you don't have it yet
2. On your Dashboard, head over to Tools > WCA Test Helper > Features and enable the `customize-store`

## Test Scenario One: The user has a theme different than TT4

1. Make sure your site has an active theme that is different than TT4
2. Access your Dashboard and head over to WooCommerce > Home
3. Click on the "Customize Your Store" button
4. Click on the "Start designing" button
5. Make sure the following modal is displayed:

<img width="1998" alt="Screenshot 2024-03-05 at 00 02 20" src="https://github.com/woocommerce/woocommerce/assets/15730971/9fe7cb79-f9d3-44ca-839c-d3d49f61c0a3">

6. Click on "Design a new theme"
7. Go through the CYS flow and as soon as the loading screen finishes loading, click on the "Done" button in the Assembler.
8. Make sure your current active theme has been changed to Twenty-twenty four.

## Test Scenario Two: The user has the TT4 theme enabled

1. Make sure your site has the TT4 theme enabled
2. Access your Dashboard and head over to WooCommerce > Home
3. Click on the "Customize Your Store" button
4. Click on the "Start designing" button
5. Ensure that you see the loading screen immediately after clicking the button (no modal should be displayed):

<img width="1997" alt="Screenshot 2024-03-05 at 00 09 39" src="https://github.com/woocommerce/woocommerce/assets/15730971/99e32ea8-f67d-4851-9872-7ee8e974856b">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Display the theme switch modal in the Customize Your Store flow only if the currently active theme is not TT4

#### Comment 

</details>
